### PR TITLE
[UserMetadata] Pass the user metadata to new executions from schedule actions

### PR DESCRIPTION
**Pull request by gow**
## What changed?
When user metadata is provided in the [ScheduleAction](https://github.com/temporalio/api/blob/4a1ab98a84b94b59f82aab4b35ea505cff968804/temporal/api/schedule/v1/message.proto#L252-L259), we were not passing it through to the new executions that were started by the scheduler. This fixes it (https://github.com/temporalio/temporal/issues/6413).

## Why?
To be feature complete. Every schedule action that contains user metadata should pass it on to the newly created executions.

## How did you test it?
Unit tests. 

## Potential risks
N/A

## Documentation
N/A

## Is hotfix candidate?
No

### CODE REVIEW

File: service/worker/scheduler/workflow.go [hunk 1]
```diff
@@ -1337,6 +1337,7 @@ func (s *scheduler) startWorkflow(
 			Header:                   newWorkflow.Header,
 			LastCompletionResult:     lastCompletionResult,
 			ContinuedFailure:         continuedFailure,
+			UserMetadata:             newWorkflow.GetUserMetadata(),
```
Comment (dnr) [inline service/worker/scheduler/workflow.go:1340]: you can just do
```suggestion
			UserMetadata:             newWorkflow.UserMetadata,
```
for consistency. if newWorkflow was nil we would already have panicked

File: service/worker/scheduler/workflow_test.go [hunk 2]
```diff
@@ -345,6 +361,7 @@ func (s *workflowSuite) TestStart() {
 		s.Equal(`"value"`, payload.ToString(req.Request.SearchAttributes.IndexedFields["myfield"]))
 		s.Equal(`"myschedule"`, payload.ToString(req.Request.SearchAttributes.IndexedFields[searchattribute.TemporalScheduledById]))
 		s.Equal(`"2022-06-01T00:15:00Z"`, payload.ToString(req.Request.SearchAttributes.IndexedFields[searchattribute.TemporalScheduledStartTime]))
+		s.Equal(userMetadata, req.Request.GetUserMetadata())
```
Comment (dnr) [inline service/worker/scheduler/workflow_test.go:364]: I think you should use [protoassert.ProtoEqual](https://github.com/temporalio/temporal/blob/main/common/testing/protoassert/assert.go#L49) here

Comment (gow) [reply service/worker/scheduler/workflow_test.go]: Done.

File: service/worker/scheduler/workflow.go [hunk 3]
```diff
@@ -1337,6 +1337,7 @@ func (s *scheduler) startWorkflow(
 			Header:                   newWorkflow.Header,
 			LastCompletionResult:     lastCompletionResult,
 			ContinuedFailure:         continuedFailure,
+			UserMetadata:             newWorkflow.UserMetadata,
```
Comment (dnr) [inline service/worker/scheduler/workflow.go:1340]: I'm assuming local activity inputs are not required to match for determinism, right?

**Review by dnr — APPROVED** [review]

File: service/worker/scheduler/workflow.go (hunk 3 above)
Comment (gow) [reply service/worker/scheduler/workflow.go:1340]: Yes. Dug a bit around and also confirmed with @cretz . This change should be good.

**Pull request merged**